File: xadapter-sample/app/src/main/java/com/xadapter/sample/MainPresenterImpl.java
package com.xadapter.sample;

import io.reactivex.network.RxNetWork;

/**
 * by y.
 * <p>
 * Description:
 */
public class MainPresenterImpl implements MainPresenter {

    private final MainView mainView;

    MainPresenterImpl(MainView mainView) {
        this.mainView = mainView;
    }

    @Override
    public void onNetRequest(int page, int type) {
        RxNetWork
                .getInstance()
                .getApi(getClass().getSimpleName(),
                        RxNetWork.observable(Api.ZLService.class)
                                .getList("daily", 20, 0), new SimpleListNetListener(mainView, type));
    }


}

File: XAdapterSample/app/src/main/java/com/xadapter/sample/MainPresenterImpl.kt
package com.xadapter.sample

import io.reactivex.network.RxNetWork

/**
 * by y.
 *
 *
 * Description:
 */
class MainPresenterImpl internal constructor(private val mainView: MainView) : MainPresenter {

    override fun onNetRequest(page: Int, type: Int) {
        RxNetWork
                .instance
                .getApi(javaClass.simpleName,
                        RxNetWork.observable(Api.ZLService::class.java)
                                .getList("daily", 20, 0), SimpleListNetListener(mainView, type))
    }
}
